p2p/gossip/protocol.go

```diff
@@ -147,16 +150,61 @@ func (prot *Protocol) markMessage(h hash) bool {
 	} else {
 		prot.Log.Debug("message is already old, hash %v", h)
 	}
-	prot.Log.Debug("marking message as old, hash %v, is already old %v", h, ok)
 	prot.oldMessageMu.Unlock()
 	return ok
 }
 
-func (prot *Protocol) propagateMessage(msg []byte, h hash) {
+type Validity int
+const (
+	Valid Validity = iota
+	Invalid
+	Unknown
+)
+
+func (prot *Protocol) isMessageValid(h hash) Validity {
+	prot.oldMessageMu.RLock()
+	res, ok := prot.invalidMessageQ[h]
+	prot.oldMessageMu.RUnlock()
+	if ok {
```

Comment:
imo this looks better : 
```suggestion
	if !ok {
	 return Unknown
	 }
	 if !res {
	  return Invalid
	 }
	 
	 return Valid
```

Reply:
I agree, done